src/dodal/common/beamlines/device_factory.py

```diff
@@ -0,0 +1,112 @@
+from collections.abc import Callable
+from dataclasses import dataclass
+from typing import Generic, ParamSpec, TypeVar
+
+from bluesky.run_engine import call_in_bluesky_event_loop
+from ophyd_async.core import DEFAULT_TIMEOUT, Device
+
+from dodal.common.beamlines.beamline_utils import ACTIVE_DEVICES
+
+P = ParamSpec("P")
+T = TypeVar("T", bound=Device)
+_skip = bool | Callable[[], bool]
+
+
+@dataclass
+class DeviceInitializationConfig:
```

Comment:
Just a consideration, but it we fold this config into the init params of DeviceInitializationController and expose them as `property`s, we could expose it in `help(dff)` to show the default values.

Reply:
`You shouldn't need to know to do help(det._config) or help(det.config),`

with this you should see the _config I guess as just a field of some type, and can explore deeper with help(det.config).

please consider making it into another issue

Reply:
https://github.com/DiamondLightSource/dodal/pull/597#pullrequestreview-2253952345

This is already a requirement for Dom.

Reply:
For the sake of velocity I'm happy for the requirement to be put in another issue if it is too hard

Reply:
https://github.com/DiamondLightSource/dodal/issues/780

Reply:
>  Just a consideration, but it we fold this config into the init params of DeviceInitializationController and expose them as propertys, we could expose it in help(dff) to show the default values.

@DiamondJoseph  so by  'fold' you mean cancel the dataclass, add it flat to the set of DeviceInitalization constructor, so that it has nearly 10 arguments, add them as private properties, then expose through `at property` getter, right?

Reply:
@stan-dot I'm happy to leave it for #780 

There's 5 values in the DeviceInitializationConfig. If we can access them in the help message it doesn't matter to me if they're on the config class or in the DeviceInitializationController directly.

But currently we're constructing and passing in a dataclass then exposing properties that are the fields of the dataclass: the intermediary class isn't being passed between objects or serialised, they are only fields of the Controller instance. The constructor goes from 1 real arg to 6. The 5 args are still presented in exactly the same way to the user in the decorator.

Reply:
I did a thing, overriding the `__repr__` call